### PR TITLE
feat: Nakamoto block support for `stacks-insepct try-mine`

### PR DESCRIPTION
**Pull request by jbencin**
### Description

Add Nakamoto block support to `stacks-inspect try-mine`

### Applicable issues

- #5551
- #5541

### Additional info (benefits, drawbacks, caveats)

This works with the testnet chainstate I've been using for testing, but I'm not sure if there are any edge cases I need to specifically handle here, depending on where in the tenure the block is

### Checklist

- [ ] Test coverage for new or modified code paths
- [ ] Changelog is updated
- [ ] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
- [ ] New clarity functions have corresponding PR in `clarity-benchmarking` repo
- [ ] New integration test(s) added to `bitcoin-tests.yml`

### CODE REVIEW

File: stackslib/src/cli.rs [hunk 1]
```diff
@@ -439,93 +440,100 @@ pub fn command_try_mine(argv: &[String], conf: Option<&Config>) {
     )
     .unwrap_or_else(|e| panic!("Failed to open mempool db: {e}"));
 
-    let tip_header = NakamotoChainState::get_canonical_block_header(chain_state.db(), &sort_db)
-        .unwrap_or_else(|e| panic!("Error looking up chain tip: {e}"))
-        .expect("No chain tip found");
+    // Parent Staccks header for block we are going to mine
```
[the remark this answers is not in the document]
Comment (obycode) [reply stackslib/src/cli.rs:443]: I see you resolved this, but it looks like the typo is still there.

[the remark this answers is not in the document]
Comment (jbencin) [reply stackslib/src/cli.rs:443]: Oops I fixed it on GitHub but force-pushed over it

[the remark this answers is not in the document]
Comment (obycode) [reply stackslib/src/cli.rs:443]: Ok. After that fix, I'm good to approve! Thanks!

**Review by obycode — APPROVED** [review]
🚀

**Review by jcnelson — APPROVED** [review]
